### PR TITLE
[IMP] Added is_unique check

### PR DESCRIPTION
**Pull request by osi-scampbell**
Add is_unique field on a res.partner.relation.type form and added method to check if a relation being created or written is unique

### CODE REVIEW

File: partner_multi_relation/models/res_partner_relation_type.py [hunk 1]
```diff
@@ -73,6 +73,12 @@ class ResPartnerRelationType(models.Model):
              " that do not fit the new criteria.\n"
              "Specify how this situation should be handled.",
     )
+    is_unique = fields.Boolean(
+        string='Is Unique',
+        help="If a relation type is unique, then there cannot be 2"
+             "relations with the same parties and the same type.",
+        default=False
```
Comment (nikul-serpentcs) [inline partner_multi_relation/models/res_partner_relation_type.py]: @osi-scampbell No need to `default` False

File: partner_multi_relation/models/res_partner_relation.py [hunk 2]
```diff
@@ -154,3 +164,32 @@ def _check_relation_uniqueness(self):
...
+                    if (type_id.browse(vals.get('type_id')).is_unique
+                       or line.type_id.is_unique):
+                        count += 1
+                else:
+                    if self.type_id.name == line.type_id.name:
+                        count += 1
+                    if self.type_id.is_unique or line.type_id.is_unique:
+                        count += 1
+                if count == 3:
+                    return False
```
Comment (max3903) [inline partner_multi_relation/models/res_partner_relation.py]: @osi-scampbell Why don't you just search if the same relation already exists?

Something like:
```
left = vals['left_partner_id'] or self.left_partner_id.id
type = vals['type_id] or self.type_id.id
right = vals['right_partner_id'] or self.right_partner_id.id
res = self.search([('left_partner_id', '=', left),
                    'type_id', '=', type),
                    'right_partner_id', '=', right)])
...
```

**Review by max3903 — APPROVED** [review]
LGTM. No tests.

**Review by osimallen — APPROVED** [review]
Approved!

**Review by wolfhall — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Why not doing this with an `@api.constrains`?

**Comment by max3903** [comment]
@pedrobaeza Good point. #697

**Pull request merged**